src/utils/project_detector.rs

```diff
@@ -0,0 +1,136 @@
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+/// Criteria is a struct containing a list of file names, extensions or folder names
+/// which will be used to check is if the Module, eg. Golang, Python etc. is valid based
+/// on the current directory.
+#[derive(Default)]
+pub struct Criteria<'a> {
+    /// files is a list of file names eg. 'package.json'
+    pub files: Vec<&'a str>,
+
+    /// extensions is a list file extensions without the . eg. 'js'
+    pub extensions: Vec<&'a str>,
+
+    /// folders is a list of folder names eg. 'node_modules'
+    pub folders: Vec<&'a str>,
+}
+
+impl<'a> Criteria<'a> {
+    pub fn new() -> Criteria<'a> {
+        Criteria {
+            ..Default::default()
+        }
+    }
+
+    pub fn set_files(mut self, files: Vec<&'static str>) -> Self {
```

Comment:
I think accepting an array may be the more idiomatic and performant way to do this.
That's what's done in the `std::process::Command` args API: https://doc.rust-lang.org/src/std/process.rs.html#490-516

Reply:
Yup your right there is less overhead and we know it's static, good thinking!